### PR TITLE
Add ArtStation presence

### PR DESCRIPTION
**Pull request by kyrie25**
#### I tried to cover as most pages and subdomains as possible for this but can still possibly miss some, but these are some  examples of how the presence works.  
  
Addressing discussion #4762 
* Home page  
image  
* Viewing an artwork  
image  
* Editing profile  
image  
* Checking inbox  
image  
* Learning  
image
image
* Viewing job postings
image
* Viewing a portfolio
image

### CODE REVIEW

File: websites/A/ArtStation/presence.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    startTimestamp: browsingTimestamp
+  };
+
+  if (document.location.pathname.startsWith("/messages"))
+    presenceData.details = "Checking inbox";
+  // (not functional)
+    // else if (document.querySelector("user-profile")) {
+    // presenceData.details = "Viewing a profile";
+    // presenceData.state = document.querySelector('meta[property="og:title"]').getAttribute("content");
+  // }
```
Comment (Slowlife01) [inline websites/A/ArtStation/presence.ts:18]: Why?

Comment (kyrie25) [reply websites/A/ArtStation/presence.ts:18]: The only differentiation between a normal page and an artist's profile that I could find is this Class name part and my `querySelector` can't actually detect it and thus labels it as a normal page
image
I left it as a comment in the code hoping somebody else can find a fix for it, or I can delete it if it's not allowed/unnecessary

**Comment by kyrie25** [comment]
Added several small image for `Checking inbox`, `Viewing an artwork`, `Exploring artworks`, `Learning` and `Learning` branches, `Editing profile`, `Uploading artwork`, `Viewing portfolio`
image
image

File: websites/A/ArtStation/presence.ts (hunk 1 above)
Comment (kyrie25) [reply websites/A/ArtStation/presence.ts]: Done

**Review by Slowlife01 — DISMISSED** [review]

**Comment by kyrie25** [comment]
Added a `View Artwork` button when viewing an artwork
image

**Review by EncryptedDev — APPROVED** [review]

**Review by Slowlife01 — APPROVED** [review]

**Pull request merged**